sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Catalog.scala

```diff
@@ -104,7 +104,7 @@ class SimpleCatalog(val conf: CatalystConf) extends Catalog {
     val tableName = getTableName(tableIdent)
     val table = tables.get(tableName)
     if (table == null) {
-      throw new NoSuchTableException
+      throw new AnalysisException("Table not found: " + tableName)
```

Comment:
@marmbrus Shouldn't we have fixed this issue by marking the relation unresolved and let the analyzer generate a proper AnalysisException ? If we were to use the catalog interface for some other purpose like tooling etc  keeping the generic catalog exceptions would have been better ?